client-keycloak/src/main/java/org/apache/atlas/keycloak/client/KeycloakClient.java

```diff
@@ -0,0 +1,183 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ * <p>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.atlas.keycloak.client;
+
+import org.apache.atlas.exception.AtlasBaseException;
+import org.apache.commons.collections.CollectionUtils;
+import org.apache.commons.lang.StringUtils;
+import org.codehaus.jettison.json.JSONException;
+import org.codehaus.jettison.json.JSONObject;
+import org.jboss.resteasy.client.jaxrs.ResteasyClientBuilder;
+import org.keycloak.admin.client.Keycloak;
+import org.keycloak.admin.client.KeycloakBuilder;
+import org.keycloak.admin.client.resource.RealmResource;
+import org.keycloak.representations.idm.GroupRepresentation;
+import org.keycloak.representations.idm.RoleRepresentation;
+import org.keycloak.representations.idm.UserRepresentation;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.apache.atlas.ApplicationProperties.ATLAS_CONFIGURATION_DIRECTORY_PROPERTY;
+
+public class KeycloakClient {
+    public static final Logger LOG = LoggerFactory.getLogger(KeycloakClient.class);
+
+    private static KeycloakClient keycloakClient = null;
+    public static Keycloak keycloak = null;
+
+    public static final String KEYCLOAK_PROPERTIES = "keycloak.json";
+
+    private static String REALM_ID;
+    private static String AUTH_SERVER_URL;
+    private static String CLIENT_ID;
+    private static String CLIENT_SECRET;
+    private static String GRANT_TYPE;
+
+
+    private KeycloakClient() {
+    }
+
+    public static KeycloakClient getKeycloakClient() {
+        if (keycloakClient == null) {
+            LOG.info("Initializing Keycloak client..");
+            try {
+                initConf();
+            } catch (IOException e) {
+                LOG.error("Failed to fetch Keycloak conf {}", e.getMessage());
+            } catch (JSONException e) {
+                LOG.error("Failed to parsing Keycloak conf {}", e.getMessage());
+            } catch (Exception e) {
+                LOG.error("Failed to connect to Keycloak {}", e.getMessage());
+            }
+            init();
```

Comment:
Should we still call the init method, if there's an error in initConf?

Reply:
Fixed, throwing exception.